transactor/src/main/scala/io/mediachain/copycat/Client.scala

```diff
@@ -121,21 +122,37 @@ class Client(client: CopycatClient) extends JournalClient {
     stateListeners.foreach(_.onStateChange(Disconnected))
   }
   
+  private def recover() {
+    val cf = client.recover()
+    recoveryExecutor.submit(new Runnable {
+      def run {
+        try {
+          cf.join()
+          logger.info("Copycat session recovered")
+        } catch {
+          case e: InterruptedException => ()
```

Comment:
Just to be clear: the call to `cf.join()` can only throw two types of exceptions, `CompletionException` and `CancellationException`. If something inside the future that `cf` encompasses throws any `Exception`, then `join` will throw a  `CompletionException` with that exception accessible through `getCause`. 

This is by contrast to `.get`, which throws `ExecutionException` instead of `CompletionException` but with almost the same semantics, and `InterruptedException` as a near-equivalent of `CancellationException`. I think. This API is super confusing.